test/lib.bash

```diff
@@ -45,7 +45,55 @@ function run_e2e_tests {
     && return 1
 }
 
+function run_knative_serving_tests {
+  (
+  # Setup a temporary GOPATH to safely check out the repository without breaking other things.
+  local tmp_gopath
+  tmp_gopath="$(mktemp -d -t gopath-XXXXXXXXXX)"
+  cp -r "$GOPATH/bin" "$tmp_gopath"
+  export GOPATH="$tmp_gopath"
+
+  # Checkout the relevant code to run
+  mkdir -p "$GOPATH/src/knative.dev"
+  cd "$GOPATH/src/knative.dev" || exit
```

Comment:
This should be `|| return 42` or `|| exit 42` or `|| return $?` So it will exit with non zero exitcode